compiler/x/codegen/OutlinedInstructions.hpp

```diff
@@ -84,6 +86,11 @@ class TR_OutlinedInstructions
    // do in a TR_OutlinedInstructions what you could have done in a normal
    // in-line internal control flow region.
    //
+   // NOTE: use this constructor directly is deprecated, for new development,
+   // use TR_OutlinedGenerator instead.
+   // TR_OutlinedGenerator will setup all necessary information for gc map and
+   // exception table.
+   //
```

Comment:
Good point and I do have a plan to make the constructor `private`. I chose not change it with this PR because downstream projects may currently use this constructor, and I want to give them a time window to adopt this change without break. I'll open a follow-up PR to make this constructor `private` once we are satisfied ourselves that downstream projects have deprecated the usage of this constructor.